stan/math/gpu/cholesky_decompose.hpp

```diff
@@ -61,56 +44,87 @@ inline matrix_gpu cholesky_decompose_recursion(matrix_gpu& A,
  *  positive definite (if m has more than 0 elements)
  */
 inline matrix_gpu cholesky_decompose(matrix_gpu& A, const int min_block) {
-  auto offset = 0;
-  auto block = floor(A.rows() / 2);
-  // NOTE: The code in this section follows the naming conventions
-  // in the report linked in the docs.
-  matrix_gpu A_11(block, block);
+  if (A.rows() == 0)
+    return A;
+  matrix_gpu L(A.rows(), A.cols());
   // Repeats the blocked cholesky decomposition until the size of the remaining
-  // submatrix is smaller or equal to the block size
-  if ((offset + block) < (A.rows())) {
-    auto block_subset = A.rows() - offset - block;
-    matrix_gpu A_21(block_subset, block);
-    matrix_gpu A_22(block_subset, block_subset);
-    // Copies a block of the input A into A_11
-    A_11.sub_block(A, offset, offset, 0, 0, block, block);
+  // submatrix is smaller or equal to the minimum blocks size
+  // or a heuristic of 100.
+  // Because the max local buffer size is 1024 we also have to recurse
+  // if the row size is greater than 1024.
+  if (A.rows() < 1024 && (A.rows() <= min_block || A.rows() < 100)) {
+    try {
+      opencl_kernels::cholesky_decompose(cl::NDRange(A.rows()),
+                                         cl::NDRange(A.rows()), A.buffer(),
+                                         L.buffer(), A.rows());
+    } catch (const cl::Error& e) {
+      check_opencl_error("cholesky_decompose", e);
+    }
+    return L;
+  } else {
```

Comment:
From the comment below: Basically here is where you'd delete the else and just indent everything over to the left one.